src/e2e-test/features/servicenowsource/RunTimeWithMacros.feature

```diff
@@ -0,0 +1,187 @@
+    And Enter runtime argument value from environment variable "rest.api.endpoint" for key "restApiEndpoint"
+    And Enter runtime argument value from environment variable "pipeline.user.username" for key "username"
+    And Enter runtime argument value from environment variable "pipeline.user.password" for key "password"
+    And Run the preview of pipeline with runtime arguments
+    Then Verify the preview of pipeline is "success"
+
+  @TS-SN-RNTM-MACRO-02 @SN_SOURCE_CONFIG @SN_RECEIVING_SLIP_LINE @BQ_SINK
+  Scenario: Verify user should be able to run a pipeline when ServiceNow plugin is configured with macros
+    When Open Datafusion Project to configure pipeline
+    And Select plugin: "ServiceNow" from the plugins list as: "Source"
+    And Navigate to the properties page of plugin: "ServiceNow"
+    And Fill Reference Name
+    And Select mode as: "TABLE"
+    And Click on the Macro button of Property: "tableName" and set the value to: "tableName"
+    And Click on the Macro button of Property: "clientId" and set the value to: "clientId"
+    And Click on the Macro button of Property: "clientSecret" and set the value to: "clientSecret"
+    And Click on the Macro button of Property: "restApiEndpoint" and set the value to: "restApiEndpoint"
+    And Click on the Macro button of Property: "user" and set the value to: "username"
+    And Click on the Macro button of Property: "password" and set the value to: "password"
+    Then Validate "ServiceNow" plugin properties
+    And Close the Plugin Properties page
+    And Select Sink plugin: "BigQueryTable" from the plugins list
+    And Connect source as "ServiceNow" and sink as "BigQuery" to establish connection
+    And Navigate to the properties page of plugin: "BigQuery"
+    And Replace input plugin property: "project" with value: "projectId"
+    And Enter input plugin property: "datasetProject" with value: "datasetprojectId"
+    And Configure BigQuery sink plugin for Dataset and Table
+    Then Validate "BigQuery" plugin properties
+    And Close the Plugin Properties page
+    And Save and Deploy Pipeline
+    And Run the Pipeline in Runtime
+    And Enter runtime argument value "receiving_slip_line" for key "tableName"
+    And Enter runtime argument value from environment variable "client.id" for key "clientId"
+    And Enter runtime argument value from environment variable "client.secret" for key "clientSecret"
+    And Enter runtime argument value from environment variable "rest.api.endpoint" for key "restApiEndpoint"
+    And Enter runtime argument value from environment variable "pipeline.user.username" for key "username"
+    And Enter runtime argument value from environment variable "pipeline.user.password" for key "password"
+    And Run the Pipeline in Runtime with runtime arguments
+    And Wait till pipeline is in running state
+    And Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
```

Comment:
check that data is written correctly to sink

Reply:
This PR is actually mainly for the ServiceNow Sink e2e tests, I have just enhanced the Source and Multi Source steps based on the latest changes in the framework. I will be raising a separate for Source and Multi Source data validation after this one.